### ISSUE TITLE
C++ batchnorm CPU version training well, but validation wrong

### ISSUE THREAD

**Issue by relaxli00** [apache/incubator-mxnet#7835]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:
Windows 7, 64bit
Compiler:

Package used (Python/R/Scala/Julia):
C++, Visual Studio 2013/2015
MXNet version:
windows prebuild Yajiedesign release 20170902
Or if installed from source:
https://github.com/yajiedesign/mxnet/releases/tag/20170902
##

I'm running a simple CNN using C++ API. CPU only, no GPU. The program runs well without any batchnorm operation. Training and validation accuracy increased reasonably with increased epochs.

However, once the batchnorm was added, e.g. between the convolution and relu. Validation didn't work correctly. I can still see the training accuracy increased reasonably. However, validation accuracy didn't show any increasing trend even after long time training.

I tried on different computers. The same network runs correctly with pure Python for both networks with and without batchnorm operation. Tried GPU version with C++ also and encountering the same problem.

Any trick to use batchnorm with C++ package, esp. the CPU version?

Thanks for any help.

**Comment by Bumblebee1964** [comment]
Having same problem. Noticed that the Mathematica version 11.2 (using MxNet under the hood) is not able to reproduce my earlier modified InceptionV3 network tested in version 10. Also the my c++/windows MxNetwork seem to have this decreasing loss but almost equal loss validation rates.

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by nswamy** [comment]
@cjolivier01 Is this fixed?

**Comment by cjolivier01** [comment]
No idea. I didn't know about this issue.

**Comment by anirudh2290** [comment]
@relaxli00 @Bumblebee1964 can you provide a minimum reproducible example ?

**Comment by Bumblebee1964** [comment]
We wnt on using Python only. As MxNet moved ahead, we will try to use the c++ package again later this year. For time being, let's close it.

**Comment by Bumblebee1964** [comment]
Hi,

thanks for addressing.

We went on using Python only. As MxNet moved ahead, we will try to use the c++ package again later this year. For time being, I have closed the thread.

Regards,

Fred

------ Original Message ------

**Comment by samskalicky** [comment]
@sandeep-krishnamurthy please close this thread. Thanks!
